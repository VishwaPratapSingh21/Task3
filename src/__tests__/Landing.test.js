import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import Landing from '../components/auth/Landing';


describe('Landing component', () => {
  const wrapper = shallow(<Landing />);

  it('reanders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('calls component did mount', () => {
    sinon.spy(Landing.prototype, 'componentDidMount');
    const fullWrapper = mount(<Landing />);
    expect(Landing.prototype.componentDidMount).toHaveProperty('callCount', 1);
    Landing.prototype.componentDidMount.restore();
  });

  it('renders a clock component', () => {
    expect(wrapper.find('Clock')).toBeTruthy();
  });
});
