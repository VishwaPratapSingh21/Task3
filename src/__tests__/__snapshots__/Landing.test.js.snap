// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component reanders properly 1`] = `<Slideshow />`;
